command/job_stop.go

```diff
@@ -140,92 +140,141 @@ func (c *JobStopCommand) Run(args []string) int {
 		return 1
 	}
 
-	// Check if the job exists
-	jobs, _, err := client.Jobs().PrefixList(jobID)
-	if err != nil {
-		c.Ui.Error(fmt.Sprintf("Error deregistering job: %s", err))
-		return 1
-	}
-	if len(jobs) == 0 {
-		c.Ui.Error(fmt.Sprintf("No job(s) with prefix or id %q found", jobID))
-		return 1
+	// TODO: since we're printing errors as
+	// we receive them, we don't need a new ch
+	// errorCh := make(chan error, len(jobIDs))
+
+	statusCh := make(chan int, len(jobIDs))
+
+	var wg sync.WaitGroup
+	for _, jobID := range jobIDs {
+		jobID := jobID
+
+		wg.Add(1)
+		go func() {
```

Comment:
Wanted to use closures here to use variables from the `Run()` scope or we pass them explicitly to another function.

Reply:
At a glance I'd say this is pushing the length of funcs we like to embed as closures, but I think it's fine to stick with for now. Refactoring once it works should be safe and easy if we want.